backend/src/services/premium/enrichment/organizationEnrichmentService.ts

```diff
@@ -186,44 +184,30 @@ export default class OrganizationEnrichmentService extends LoggerBase {
 
         delete org.identities
 
-        // Check for an organization with the same website exists
-        let existingOrg
-        let orgService
-
+        // check for organization with same website and add them to merge suggestions
         if (org.website) {
-          existingOrg = await OrganizationRepository.findByDomain(org.website, this.options)
-          orgService = new OrganizationService({ ...this.options, transaction })
-        }
-
-        if (existingOrg && existingOrg.id !== org.id) {
-          await orgService.merge(existingOrg.id, org.id)
-          unmergedOrgs.push({ ...org, id: existingOrg.id })
-        } else {
-          unmergedOrgs.push(org)
-        }
-      }
+          let existingOrg = await OrganizationRepository.findByDomain(org.website, this.options)
 
-      // Check if two or more orgs in the umergedOrgs list have same website
-      const duplicateOrgs = []
-      const uniqueWebsites = new Set()
+          if (!existingOrg && existingOrgMap.has(org.website)) {
+            existingOrg = existingOrgMap.get(org.website)
+          }
 
-      for (const org of unmergedOrgs) {
-        if (uniqueWebsites.has(org.website)) {
-          duplicateOrgs.push(org)
-        } else {
-          uniqueWebsites.add(org.website)
+          if (existingOrg && existingOrg.id !== org.id) {
+            await OrganizationRepository.addToMerge(
+              [{ organizations: [org.id, existingOrg.id], similarity: null }],
+              this.options,
+            )
+            delete org.website
+          } else {
+            existingOrgMap.set(org.website, org)
```

Comment:
isn't just setting the organizationId as value enough here instead of the full org object?